types/index.d.ts

```diff
@@ -3,4 +3,4 @@
 import { DayPicker } from "./DayPicker";
 // We need to use 'export =' to be compatible with the 'module.exports = DayPicker.default || DayPicker' syntax
 // in DayPicker.js in the final NPM package.
-export = DayPicker;
+export default DayPicker;
```

Comment:
This is a breaking change, please don't make it so lightly. There is clear guidance in DefinitelyTyped's README about when to use default exports. See here: https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

As @gpbl stated, there's a reason I left the comment on the line above (which is now invalidated with this code change...).

Can you point me to the code in the distributed `react-day-picker` package which suggests that this is the right way to export `DayPicker`?

Reply:
Well it's not about right way per say, it's just it doesn't work at all with typescript. These are only definitions after all. Export default is the standard way of doing it these days. I actually never seen export = something in TS before. It's not as much suggestion as both me and @firsttris encountered a problem that appears to be fixable by changing this. Reading that comment, IMHO that source file should do something like 
export DayPicker from 'sourcefile' (afk)

In fact this is how I achieve similar functionality in one of my libs https://github.com/PeterKottas/guestbell-forms/blob/master/src/lib/index.ts with no problems so far. Obviously that's mostly used internally but we haven't encountered problems till now. Maybe I am missing something

Reply:
> it doesn't work at all with typescript

This isn't true; we use it in [blueprint](https://github.com/palantir/blueprint) with the `export =` typings. I've used them in a handful of packages with TypeScript and they work fine. Did you look at the DefinitelyTyped README section which I linked? Was any part of that not clear enough?

Again, I am still curious to the answer to my question above:

> Can you point me to the code in the distributed react-day-picker package which suggests that this is the right way to export DayPicker?

Lastly, do you acknowledge that this is a breaking change for typescript consumers, regardless of the fact that it fixes your particular tsconfig + bundling setup?

Reply:
Yeah that's cool then as long as it works for you. But the behavior we have here is inconsistent. With the documentation. Here http://react-day-picker.js.org/ it says this is the correct way of importing the lib. 

import DayPicker from 'react-day-picker';

That's not the case in typescript. Therefore I wouldn't say it's fixing my particular problem. It seems to be fixing the issue where docs are inconsistent with the way it actually works. 

Readme makes sense to me, however that doesn't change the fact that it doesn't work. 

I agree that it's a breaking change. But in my mind, it's more of a bug fix. It's fixing something that is currently not working. 

I am not sure I understand that question to be honest. @firsttris pointed out the code that is causing this problem. It's just the typings, root file for DayPicker. No sure what else are you looking for